server/src-lib/Hasura/GraphQL/Resolve/Context.hs

```diff
@@ -151,3 +152,9 @@ withSelSet selSet f =
 
 fieldAsPath :: (MonadError QErr m) => Field -> m a -> m a
 fieldAsPath = nameAsPath . _fName
+
+resolvePGCol :: (MonadError QErr m)
+             => PGColGNameMap -> G.Name -> m PGColumnInfo
+resolvePGCol colFldMap fldName =
+  onNothing (Map.lookup fldName colFldMap) $ throw500 $
+  "no column associated with name " <> G.unName fldName
```

Comment:
This is everwhere. We carry a context in a Hashmap with `item -> information`. And we expect each `item` we come across should definitely have `information` since we generate the GraphQL schema in that way and the `item` is already validated. It'll be highly useful if we found a way to avoid `throw500`s